podcast/2.17.md

```diff
@@ -0,0 +1,487 @@
+
+**Max Brunsfeld:**
+
+Yeah. It took a while. Fortunately we had this company summit where we all flew to San Diego and we stayed in a resort together and GitHub-
+
+**Beyang Liu:**
+
+How big was the company at this point?
+
+**Max Brunsfeld:**
+
+I think the number of software engineers at the company was around 150. So there were a lot of people I didn't know because I'd really only met the Atom team, which was this small island. And Rob Rix had been at GitHub for a while and I didn't know him, but he went up on stage in front of all the software engineers at the company and pitched this idea that we should be doing parsing of the code on GitHub and trying to build productivity features that made use of the parse tree. After he gave that talk I had to go and find him and say, "Hey I really think we should do that too. And I've been working on this parsing framework."
+
+My focus was on text editors at the time, but it was kind of a lot of common design constraints for a text editor and also for github.com because they're both these tools where we, the tool creators, don't know a ton about the code that the person is viewing with the tool. We just know the file extension and there's a lot of different languages that we want to be able to plug in. And the parsing needs to be really fast in an editor because your display is updating while you're typing and you need to see that. But in GitHub, it's just because there's a lot of pushes per second and so the parsing needs to be really fast.
+
+And so I talked to him and he kind of got set up with Tree-sitter and they were doing code analysis in Haskell. And so they quickly created a Tree-sitter binding for Haskell and they explored a bunch of ideas doing stuff with diffs, but what eventually shipped was this first version of code navigation in GitHub. It was still simple. It was similar to Ctags where it's not doing a semantic analysis of the code, but it's finding functions and finding classes and types and finding call sites to functions and matching them up by their name. It was at least better than Ctags. Also, you could find not just the definition, but the call sites as well–although newer versions of Ctags might do that as well.
+
+**Beyang Liu:**
+
+This was back in like 2015 or something like that?
+
+**Max Brunsfeld:**
+
+Yeah, I started working at GitHub at the end of 2014. I think this might have been 2015 or 2016 when they started working on this. I was going to be involved from afar. I would help them debug Tree-sitter stuff and they were working on grammars that made a bunch more grammars for Ruby and PHP and Python.
```

Comment:
Is "working on grammars that made a bunch more grammars" accurate, or should the the "that" be an "and"?